packages/create-sitecore-jss/src/templates/vue/scripts/generate-config.js

```diff
@@ -45,7 +49,16 @@ const config = {};\n`;
 
   // Set base configuration values, allowing override with environment variables
   Object.keys(config).forEach((prop) => {
-    configText += `config.${prop} = process.env.VUE_APP_${constantCase(prop)} || "${config[prop]}",\n`;
+    if (prop === 'siteName') {
+      // for the sake of backwards compatibility - make sure to initialize siteName from jssAppName if necessary
+      configText += `config.${prop} = process.env.VUE_APP_${constantCase(prop)} || "${
+        config[prop]
+      } || process.env.VUE_APP_JSS_APP_NAME || ${config.jssAppName}",\n`;
+    } else {
+      configText += `config.${prop} = process.env.VUE_APP_${constantCase('jssAppName')} || "${
```

Comment:
I think this is an error - if prop is not siteName, we assign a jssAppName value to it. But there could be other non-siteName props.
In general, perhaps it makes sense to not make a special case for siteName and instead use what you already have on lines 38-39:
`config.siteName = config.siteName || config.jssAppName`
or use a value from process.env as default for siteName on line 21, similar to how it's done in nextjs template

Reply:
Yea, I added that ugly special case because on line 38 jssAppName could be not initialized yet. Initializing both in defaultConfig using process.env values should solve this, thanks.